# Plastid genome of White spruce (*Picea glauca*)
:evergreen_tree: Annotation of the plastid genome of white spruce (*Picea glauca*)
[https://www.ncbi.nlm.nih.gov/nuccore/MK174379](https://www.ncbi.nlm.nih.gov/nuccore/MK174379)

![WS77111 chloroplast genome](https://github.com/bcgsc/picea-glauca-plastid/blob/master/ogdraw/ogdraw-WS77111-chloroplast-outfile-preview.png)

## Methods and Results
The white spruce WS77111 chloroplast assembly was annotated using [GeSeq](https://chlorobox.mpimp-golm.mpg.de/geseq.html). The GenBank file generated by GeSeq was then converted into a Gene Feature File using [EMBOSS Seqret](https://www.ebi.ac.uk/Tools/sfc/emboss_seqret/), where duplicate annotations were removed and manual annotations were added. Reference chloroplast genomes used include interior spruce PG29 and Sitka spruce Q903, and occasionally the Norway spruce. In addition to GeSeq, two third party tRNA annotators were used: [tRNAscan-SE v2.0](http://lowelab.ucsc.edu/tRNAscan-SE/) and [ARAGORN v1.2.38](http://mbio-serv2.mbioekol.lu.se/ARAGORN/). Although these third party tRNA annotators did in fact detect some 'novel' tRNAs, these tRNAs were not found in all reference chloroplast genomes used. Further analysis of these tRNAs was conducted using [RNAweasel](http://megasun.bch.umontreal.ca/cgi-bin/RNAweasel/RNAweaselInterface.pl), and ARAGORN to produce 2D structures and folding results of the tRNAs. Due to these inconclusive results and the fact that the spruce chloroplast genome is known to be highly conserved, it was concluded that these tRNAs to be excluded in the final annotation. Inverted repeats were also found but excluded from the final annotation. 

## GeSeq
The assembled FASTA file was inputted into [GeSeq](https://chlorobox.mpimp-golm.mpg.de/geseq.html).
* see [Documentation](https://academic.oup.com/nar/article/45/W1/W6/3806659)
* [GeSeq Settings](https://github.com/bcgsc/picea-glauca-plastid/blob/master/GeSeq/GeSeq_settings.png)
* GeSeq generates an [OGDRAW .png file](https://github.com/bcgsc/picea-glauca-plastid/blob/master/GeSeq/WS77111-chloroplast.png) and a [GenBank .gb file](https://github.com/bcgsc/picea-glauca-plastid/blob/master/GeSeq/WS77111-chloroplast-GenBank.gb).

## EMBOSS Seqret
The .gb file is converted to a [.gff file](https://github.com/bcgsc/picea-glauca-plastid/blob/master/WS77111-chloroplast-EMBOSS.gff) using [EMBOSS Seqret](https://www.ebi.ac.uk/Tools/sfc/emboss_seqret/):
* [EMBOSS Seqret settings](https://github.com/bcgsc/picea-glauca-plastid/blob/master/EMBOSS-setings.png)

## Manual Annotation
Duplicates were removed. Most conflicts were due to the Picea morrisonicola and Picea asperata reference annotations. One of them annotated tRNAs with anti-codons, and the other did not, so they were treated as different annotations and placed in the GeSeq generated .gb file twice. Those without anti-codons were removed from the final file. 

ARAGORN and tRNAscan detected some tRNAs that GeSeq did not detect. However, those were not detected with high confidence and were removed from the final annotation as they were not present in PG29, Q903, or Norway spruce annotations, making them highly unlikely as most of them are highly conserved sequences (see [diagram](https://github.com/bcgsc/picea-glauca-plastid/blob/master/VennDiagram.pdf).
* GeSeq files were regenerated with and without third party tRNA annotators to cross reference which annotations were valid:
	* with [only ARAGORN](https://github.com/bcgsc/picea-glauca-plastid/tree/master/GeSeq/ARAGORN-only), no tRNAscan
	* with [only tRNAscan](https://github.com/bcgsc/picea-glauca-plastid/tree/master/GeSeq/tRNAscan-only), no ARAGORN
	* with [only GeSeq](https://github.com/bcgsc/picea-glauca-plastid/tree/master/GeSeq/GeSeq-only) (without both ARAGORN and tRNAscan)

Four genes specifically needed to be manually annotated: rps12, petB, petD, rpl16. Rps12 is trans-spliced, while the other genes had initial short exons. 

GeSeq did not annotate some mRNAs as well as some exons, which were later manually annotated as well (see final annotation). In the final annotation, all 114 genes were conserved, including the 74 coding regions (CDS), 4 rRNAs, 36 tRNAs, and 15 introns (9 of them in coding regions, 6 in tRNAs).

### RNAweasel 
[RNAweasel](http://megasun.bch.umontreal.ca/cgi-bin/RNAweasel/RNAweaselInterface.pl) used to confirm that the tRNAs were not valid, but tRNA-Ser was worth looking into further.
* [RNAweasel settings](https://github.com/bcgsc/picea-glauca-plastid/blob/master/RNAweasel/RNAweasel-settings.png)
* [RNAweasel results](https://github.com/bcgsc/picea-glauca-plastid/blob/master/RNAweasel/RNAweasel-results.txt)

### ARAGORN
[ARAGORN](http://mbio-serv2.mbioekol.lu.se/ARAGORN/) was run independently of GeSeq to generate the ARAGORN text report with 2D tRNA structures. The tRNA in question tRNA-Ser is tRNA #17.
* [ARAGORN settings](https://github.com/bcgsc/picea-glauca-plastid/blob/master/ARAGORN-settings.png)
* [ARAGORN results](https://github.com/bcgsc/picea-glauca-plastid/blob/master/ARAGORN-tRNA.txt)

### tRNAscan
[tRNAscan](http://lowelab.ucsc.edu/tRNAscan-SE/), in conjunction with ARAGORN, was used to determine tRNA products.

## table2asn_GFF
GFF annotation validated using [table2asn_GFF](https://www.ncbi.nlm.nih.gov/genbank/genomes_gff):
* Generated Files: [Sequin ASN.1 file](https://github.com/bcgsc/picea-glauca-plastid/blob/master/table2asn_GFF/WS77111-chloroplast.sqn), [discrepancy report](https://github.com/bcgsc/picea-glauca-plastid/blob/master/table2asn_GFF/WS77111-chloroplast.dr), [error list](https://github.com/bcgsc/picea-glauca-plastid/blob/master/table2asn_GFF/WS77111-chloroplast.val), [Genbank file](https://github.com/bcgsc/picea-glauca-plastid/blob/master/table2asn_GFF/WS77111-chloroplast.gbf)
* See [Makefile](https://github.com/bcgsc/picea-glauca-plastid/blob/master/table2asn_GFF/Makefile)

## OGDRAW
The .gbf file generated by table2asn_GFF was fed through [OGDRAW](https://chlorobox.mpimp-golm.mpg.de/OGDraw.html):
* [OGDraw settings](https://github.com/bcgsc/picea-glauca-plastid/blob/master/ogdraw/OGDRAW_settings.png)
* OGDraw files: see [ogdraw](https://github.com/bcgsc/picea-glauca-plastid/tree/master/ogdraw)

## Citation
Lin D, Coombe L, Jackman SD, Gagalova KK, Warren RL, Hammond SA, Kirk H, Pandoh P, Zhao Y, Moore RA, Mungall AJ, Ritland C, Jaquish B, Isabel N, Bousquet J, Jones SJM, Bohlmann J, Birol I. 2019. Complete chloroplast genome sequence of a white spruce (*Picea glauca*, genotype WS77111) from eastern Canada. [Microbiol Resour Announc 8:e00381-19. doi: 10.1128/MRA.00381-19.](https://mra.asm.org/content/8/23/e00381-19)
