# picea-glauca-plastid
:evergreen_tree: Annotate the plastid genome of white spruce (Picea glauca)

## GeSeq
The assembled FASTA file was inputted into [GeSeq](https://chlorobox.mpimp-golm.mpg.de/geseq.html).
* see [Documentation](https://academic.oup.com/nar/article/45/W1/W6/3806659)

GeSeq Settings: see GeSeq_settings.png
* GeSeq generates an OGDRAW .png file and a GenBank .gb file.

## EMBOSS Seqret
The .gb file is converted to a .gff file using [EMBOSS Seqret](https://www.ebi.ac.uk/Tools/sfc/emboss_seqret/):
* EMBOSS Seqret settings: see EMBOSS-settings.png

## Manual Annotation
Duplicates were removed. Most conflicts were due to the Picea morrisonicola and Picea asperata reference annotations. One of them annotated tRNAs with anti-codons, and the other did not, so they were treated as different annotations and placed in the GeSeq generated .gb file twice. Those without anti-codons were removed from the final file. 

ARAGORN and tRNAscan detected some tRNAs that GeSeq did not detect. However, those were not detected with high confidence and were removed from the final annotation as they were not present in PG29, Q903, or Norway spruce annotations, making them highly unlikely as most of them are highly conserved sequences (see VennDiagram.pdf).
* GeSeq files were regenerated with and without third party tRNA annotators to cross reference which annotations were valid:
	* with only ARAGORN, no tRNAscan
	* with only tRNAscan, no ARAGORN
	* without both  ARAGORN and tRNAscan
* see these files in GeSeq

Four genes specifically needed to be manually annotated: rps12, petB, petD, rpl16. Rps12 is trans-spliced, while the other genes had initial short exons. 

GeSeq did not annotate some mRNAs as well as some exons, which were later manually annotated as well (see final annotation). In the final annotation, all 114 genes were conserved, including the 74 coding regions (CDS), 4 rRNAs, 36 tRNAs, and 15 introns (9 of them in coding regions, 6 in tRNAs).

### RNAweasel 
[RNAweasel](http://megasun.bch.umontreal.ca/cgi-bin/RNAweasel/RNAweaselInterface.pl) used to confirm that the tRNAs were not valid, but tRNA-Ser was worth looking into further.
* RNAweasel settings: see RNAweasel-settings.png 
* RNAweasel results: see RNAweasel-results.txt

### ARAGORN
[ARAGORN](http://mbio-serv2.mbioekol.lu.se/ARAGORN/) was run independently of GeSeq to generate the ARAGORN text report with 2D tRNA structures. The tRNA in question tRNA-Ser is tRNA #17.
* ARAGORN settings: see ARAGORN-settings.png
* ARAGORN results: see ARAGORN-tRNA.txt

## OGDRAW
The .gb file generated by GeSeq was later manually re-edited to match changes made in the .gff file. This newly edited .gb file was fed through [OGDRAW](https://chlorobox.mpimp-golm.mpg.de/OGDraw.html):
* OGDraw settings: see OGDRAW_settings.png
* OGDraw files: see ogdraw

## table2asn
GFF annotation validated using [table2asn_GFF](https://www.ncbi.nlm.nih.gov/genbank/genomes_gff):
* Generated files: WS77111-chloroplast.sqn WS77111-chloroplast.dr WS77111-chloroplast.val WS77111-chloroplast.stats WS71111-chloroplast.gbf
* See Makefile
